JobSchedulerLab/app/src/main/java/com/justinwells/jobschedulerlab/MainActivity.java

```diff
@@ -0,0 +1,91 @@
+package com.justinwells.jobschedulerlab;
+
+import android.app.job.JobInfo;
+import android.app.job.JobScheduler;
+import android.content.ComponentName;
+import android.os.PersistableBundle;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+public class MainActivity extends AppCompatActivity {
+    public static final int JOB_ONE_ID = 1;
+    public static final int JOB_TWO_ID = 2;
+    public static final int JOB_THREE_ID = 3;
+
+
+    private static TextView oldTime, newTime, oldColor, newColor;
+    private static ImageView battery;
```

Comment:
This works for this lab, but it's a really bad idea. Views always have a reference to their Activity, and static variables remain in memory for the life of the application. This means that via these static variables, you have a reference to MainActivity that will never go away, so MainActivity can never be garbage-collected. This is a memory leak.

Since this app only has one activity anyway, it won't hurt anything. But in a bigger app with multiple activities, you need to be really careful to never hold a reference to an activity beyond that activity's normal lifecycle.